src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -389,10 +389,9 @@ private void RemoveFromStats(Stats current)
             {
                 for (int i = 0; i < _allStats.Count; i++)
                 {
-                    if (_allStats[i].TryGetTarget(out Stats? stats) && stats == current)
+                    if (_allStats[i].TryGetTarget(out Stats? stats))
```

Comment:
This is going to remove all elements with non-null target from the collection and the only elements left in the collection are going to be the ones with null target. Is it the desired behavior?

Reply:
Good catch, fixed. I didn't add any test because they are complex to write givin this code path is in a finalizer.